.github/workflows/deploy-gh-pages.yaml

```diff
@@ -17,35 +17,103 @@ permissions:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        service:
+          - account
+          - content-watcher
+          - content-publishing
+          - graph
+
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - name: Set up mdBook 📚
+        if: matrix.service == 'account' # Only upload mdbook once
         uses: ./.github/workflows/common/set-up-mdbook
         with:
           token: ${{secrets.GITHUB_TOKEN}}
       - name: Setup Pages
+        if: matrix.service == 'account' # Only setup Pages once
         id: pages
         uses: actions/configure-pages@v4
 
       - name: Build with mdBook
+        if: matrix.service == 'account' # Only build mdbook once
         working-directory: docs
         run: mdbook build && ls book
 
-      - name: Upload artifact
-        uses: actions/upload-pages-artifact@v3
+      # Upload the mdBook output to GitHub Pages
+      - name: Upload mdbook to GitHub Pages
+        if: matrix.service == 'account' # Only upload mdbook once
+        uses: actions/upload-artifact@v4
         with:
-          path: ./docs/book
+          name: mdbook
+          path: ./docs
```

Comment:
Why not make this a separate job & cache the output, if it only needs to be done once, then publish at the end? Checking for 'account' on multiple steps seems hokey...